agent/internal/agent.go

```diff
@@ -124,29 +122,14 @@ func (a *Agent) run(ctx context.Context) error {
 	var cruntime container.ContainerRuntime
 	switch a.opts.ContainerRuntime {
 	case options.PodmanContainerRuntime:
```

Comment:
I think we also want to remove this switch statement; no one should even be using the configuration option. 

I think we should simply set `cruntime` to `options.DockerContainerRuntime` (or whatever makes sense in the code). We can check if `a.opts.ContainerRuntime` has a value and is not docker if we want to print a generic warning ... but I think we can simplify this section of the code.

Reply:
Yes, earlier i removed the switch statement all together and had only Docker configs, but @NicholasBlaskey suggested we throw error in [here](https://github.com/determined-ai/determined/pull/9516#discussion_r1638781305)

I agree to both cases, maybe now we can return an error, and later just have a `case default` which handles with a generic error message

Reply:
This configuration option shouldn't be known to users; we do not want them using it even if they have discovered it.

It looks like docker is the default for container runtime configuration. I think this because there's no `default` clause in the original switch statement, but you should confirm this is true.

I think we can have a much simpler check. In pseudo-code, something like:
```
if a.opts.ContainerRuntime != "" and a.opts.ContainerRuntime != "Docker" {
  // print warning
  // error and exit
}

// continue to set up ContainerRuntime interface using Docker
```

Let me know if this is still confusing and we can go over the code together!